apps/studio/src/components/ResourceSelector/ResourceSelectorContent.tsx

```diff
@@ -0,0 +1,150 @@
+  hasAdditionalLeftPadding: boolean
+  handleClickResourceItem: (
+    resourceItemWithAncestryStack: ResourceItemContent[],
+  ) => void
+}) => {
+  return resourceItemsWithAncestryStack.map((resourceItemWithAncestryStack) => {
+    const lastChild = lastResourceItemInAncestryStack(
+      resourceItemWithAncestryStack,
+    )
+
+    if (!lastChild) {
+      throw new Error(
+        "Unexpected undefined lastChild from lastResourceItemInAncestryStack",
+      )
+    }
+
+    return (
+      <ResourceItem
+        key={lastChild.id}
+        item={lastChild}
+        isDisabled={isResourceItemDisabled(lastChild)}
+        isHighlighted={isResourceIdHighlighted(lastChild.id)}
+        handleOnClick={() =>
+          handleClickResourceItem(resourceItemWithAncestryStack)
+        }
+        hasAdditionalLeftPadding={hasAdditionalLeftPadding}
+      />
+    )
+  })
+}
+
+const ZeroResult = ({
+  searchQuery,
+  handleClickClearSearch,
+}: {
+  searchQuery: string
+  handleClickClearSearch: () => void
+}) => {
+  return (
+    <VStack
+      h="full"
+      w="full"
+      alignItems="center"
+      justifyContent="center"
+      gap="0.75rem"
+    >
+      <VStack
+        w="full"
+        alignItems="center"
+        justifyContent="center"
+        gap="0.25rem"
+      >
+        <Text textStyle="subhead-2" textAlign="center">
+          We can't find anything with
+          <br />"{searchQuery}" in title
+        </Text>
+        <Text textStyle="caption-2">Try searching for something else.</Text>
+      </VStack>
+      <Button variant="link" size="xs" onClick={handleClickClearSearch}>
+        <Text textStyle="caption-2">Clear search</Text>
+      </Button>
+    </VStack>
+  )
+}
+
+export const LoadingResourceItemsResults = () => {
+  return Array.from({ length: 5 }).map((_, index) => (
+    <ResourceItemSkeleton key={`loading-${index}`} />
+  ))
+}
+
+export const SuspendableContent = ({
+  resourceItemsWithAncestryStack,
+  isResourceIdHighlighted,
+  isResourceItemDisabled,
+  hasAdditionalLeftPadding,
+  handleClickResourceItem,
+  isSearchQueryEmpty,
+  searchQuery,
+  clearSearchValue,
+  isLoading,
+}: {
+  resourceItemsWithAncestryStack: ResourceItemContent[][]
+  isResourceIdHighlighted: (resourceId: string) => boolean
+  isResourceItemDisabled: (resourceItem: ResourceItemContent) => boolean
+  hasAdditionalLeftPadding: boolean
+  handleClickResourceItem: (
+    resourceItemWithAncestryStack: ResourceItemContent[],
+  ) => void
+  isSearchQueryEmpty: boolean
+  searchQuery: string
+  clearSearchValue: () => void
+  isLoading: boolean
+}) => {
+  if (isLoading) {
+    return <LoadingResourceItemsResults />
+  }
+  if (resourceItemsWithAncestryStack.length === 0) {
+    return isSearchQueryEmpty ? (
+      <NoItemsInFolderResult />
+    ) : (
+      <ZeroResult
+        searchQuery={searchQuery}
+        handleClickClearSearch={clearSearchValue}
+      />
+    )
+  }
```

Comment:
style (not actionable): should we prefer doing early returns rather than having the nested ternary for readability?

```tsx
const hasNoItems = resourceItemsWithAncestryStack.length === 0
if (hasNoItems && isSearchQueryEmpty) return ...
if (hasNoItems) return ...
```

Reply:
fixed in https://github.com/opengovsg/isomer/pull/865/commits/97de6fef8fd11fba128e8928d71cea07df2df357